### PR TITLE
Issues #45 #44 - Removed delete, works with _super, refactored. (NEEDS REVIEW)

### PR DESCRIPTION
**Pull request by michaelBenin**
Issue #45 - removed delete, refactored plugin
Issue #44 - fnTest is way too broad

This commit refactored the project and fixed one bug, where
naming a dataType _super caused issues. See: https://github.com/lukasolson/backbone-super/issues/44

Also we refactored the structure of this project. One thing I'd like some feedback on is using the `for in` loop. I'm curious about the projects status working
x browser.

- Fix: Removed delete
- Fix: allowed for dataTypes to be named _super
- Add: Documentation, refactoring

TODO: Perf testing before and after, add more tests that run in browser as
that is most likely where this library is used.

### CODE REVIEW

**Comment by michaelBenin** [comment]
This line specifically:

https://github.com/michaelBenin/backbone-super/blob/master/backbone-super/backbone-super.js#L58

What browsers are we expected to support?

**Comment by michaelBenin** [comment]
If this makes it in, even though this is all backwards compatible, I'd still like to see a major version bump.

**Comment by michaelBenin** [comment]
@philfreo could you test your app with this version?

**Comment by philfreo** [comment]
It seems like this breaks the ability for subclasses of Backbone.Router to call `this.listenTo`

File: backbone-super/backbone-super.js [hunk 1]
```diff
@@ -1,111 +1,128 @@
...
+    // later.
+    child.__super__ = parent.prototype;
+
+    return child;
+  }
 
-		// Set a convenience property in case the parent's prototype is needed later.
-		child.__super__ = parentProto;
+  // Set up inheritance for the model, collection, router, view and history.
+  Backbone.Model.extend = Backbone.Collection.extend = Backbone.Router.extend = Backbone.View.extend = Backbone.History.extend = extend;
```
Comment (michaelBenin) [inline backbone-super/backbone-super.js]: Remove Backbone.Router as well as Backbone.History as original backbone-super did not modify these namespaces.

**Comment by michaelBenin** [comment]
The original backbone-super didn't modify the router, see comment here:

https://github.com/lukasolson/backbone-super/pull/46/files#diff-6ff8edd566a457378be5ae7e58c16c6aR123

I'll remove modifying the Router and History namespace on Backbone.

**Comment by michaelBenin** [comment]
Thank you @philfreo

**Comment by michaelBenin** [comment]
@philfreo - this pr has been updated with the fix for the execution scope/context in the makeSuper method. The listenTo's are fixed. Ran backbone's test suite with this updated backbone-super and all passes. This repo could be updated to run the tests from backbone with backbone super as I did it manually. Creating an issue for that future fix.

**Comment by michaelBenin** [comment]
Also see here on information regarding testing this project with backbone's tests:

https://github.com/lukasolson/backbone-super/issues/47

**Comment by michaelBenin** [comment]
@philfreo if you get time this week, could you test your use case with the updated pr?

**Comment by philfreo** [comment]
I'm getting a new error now... some code that's always worked fine (w/the original backbone-super) now gives an error. Here's a screenshot of the relevant parts inside the debugger (`this` is a view instance, `this.formClass` is supposed to refer to another View)

screenshot 2015-03-01 10 37 25

**Comment by philfreo** [comment]
After this latest commit I'm getting all sorts of warnings & errors regarding [Backbone-Relational](https://github.com/PaulUithol/Backbone-relational). Haven't looked into the details but pretty sure something else is wrong with this refactor.

**Comment by michaelBenin** [comment]
Sounds good we can table this.

**Pull request closed without merging**